test/e2e/suite_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/test/e2e/framework"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	k8sClient client.Client
+	ctx       context.Context
+)
+
+func TestAPIs(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t,
+		"End To End Suite",
+	)
+}
+
+var _ = ginkgo.BeforeSuite(func() {
+	k8sClient = framework.CreateClientUsingCluster()
+	ctx = context.Background()
+	framework.KueueReadyForTesting(k8sClient)
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	kindName := os.Getenv("KIND_CLUSTER_NAME")
+	if kindName == "" {
+		kindName = "kind"
+	}
+	cmd := exec.Command("kind", "delete", "cluster", "--name", kindName)
+	gomega.Expect(cmd.Run()).Should(gomega.Succeed())
```

Comment:
Nope... leave it to the user to do outside of the test.

We are not creating the cluster in the suite, so we shouldn't delete it either.

Reply:
I'm not really sure what to do here.  @BenTheElder has pointed out that we should probably handle creation and deletion in a bash script that calls this e2e test.  

The test-infra project typically follows this pattern where the Makefile calls a shell script that does creation and deletion of any objects you need for the test.  

Projects like the descheduler also follow a similar pattern: https://github.com/kubernetes-sigs/descheduler/blob/master/test/run-e2e-tests.sh

I followed the pattern for using the Makefile because we don't use any bash scripts for testing in this project so I wanted to keep a similar pattern.

Reply:
We do have a few scripts. I think it makes a lot of sense in this case because we need the trap

Reply:
Okay.  So I should create a similar bash script that does creation and deletion of a kind cluster.  And it should delete even if test fails.  Both the test-infra PR and local runs should assume creation of a cluster and deletion of a cluster with test.  Maybe a followup can be making this smarter so that someone could use an existing Kind cluster for faster testing? Or do we want to use existing cluster as an option as part of this PR?

And I'll follow up with the other items you requested also.  Just want some clarity on this point before I code it.

Does that sound like a good plan?

Reply:
The script can be named something like: `run-e2e-tests-on-new-kind-cluster.sh`

Whereas `make test-e2e` should be able to run on an existing cluster.

Reply:
Sounds good.